cyder/core/cyuser/backends.py

```diff
@@ -58,37 +58,52 @@ def _has_perm(user, ctnr, action, obj=None, obj_class=None):
     if user.is_superuser:
         return True
 
+    ctnr_level = -1
+    assert LEVEL_ADMIN > LEVEL_USER > LEVEL_GUEST > ctnr_level
+
+    if obj:
+        try:
+            ctnrs = obj.get_ctnrs()
+        except TypeError:
+            ctnrs = None
```

Comment:
Yes, that is the intended behavior. Nameservers have no container, so the user's active container is irrelevant. This is why `ctnr` is set to `None`. Check out the line below... `if obj and ctnr and not ctnr.check_contains_obj(obj):`. If the active container was used, this would cause the permissions check to abort in the case of a Nameserver, because the active container does not contain the Nameserver. Nameserver only cares about `is_cyder_guest` and `is_cyder_admin`, really.

I understand that this code is a bit convoluted... but I think that's the nature of checking permissions this way. I tried to do what I could to make the permissions system do what we want it to, without changing too many things that might break something else.